### PR TITLE
Port base resource item classes from tools/base/resource-repository

### PR DESCRIPTION
**Pull request by kevinzheng-ap**
Part of https://github.com/cashapp/paparazzi/issues/524

keep them under `resource/base` for now to prevent conflicts with existing `BasicResourceItem`

cc @jrodbx

### CODE REVIEW

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItem.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.android.ide.common.rendering.api.ResourceValue
+import com.android.ide.common.resources.ResourceItemWithVisibility
+
+/**
+ * Ported from: [BasicResourceItem.java](https://cs.android.com/android-studio/platform/tools/base/+/47d204001bf0cb6273d8b135c7eece3a982cf0e0:resource-repository/main/java/com/android/resources/base/BasicResourceItem.java)
+ *
+ * Combination of [ResourceItemWithVisibility] and [ResourceValue] interfaces.
+ */
+interface BasicResourceItem : ResourceItemWithVisibility, ResourceValue
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItem.kt:26]: delete?  this is app.cash.paparazzi.internal.resources.BasicResourceItem!

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItemBase.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import com.android.utils.HashCodes
+import com.google.common.base.MoreObjects
+
+/**
+ * Ported from: [BasicResourceItemBase.java](https://cs.android.com/android-studio/platform/tools/base/+/18047faf69512736b8ddb1f6a6785f58d47c893f:resource-repository/main/java/com/android/resources/base/BasicResourceItemBase.java)
+ *
+ * Base class for implementations of the [BasicResourceItem] interface.
+ *
+ */
+abstract class BasicResourceItemBase(
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItemBase.kt:33]: delete?  BasicResourceItemBase + BasicResourceItem on AOSP == app.cash.paparazzi.internal.resources.BasicResourceItem

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * Ported from: [BasicValueResourceItemBase.java](https://cs.android.com/android-studio/platform/tools/base/+/18047faf69512736b8ddb1f6a6785f58d47c893f:resource-repository/main/java/com/android/resources/base/BasicValueResourceItemBase.java)
+ *
+ * Base class for value resource items.
+ */
+abstract class BasicValueResourceItemBase(
+  type: ResourceType,
+  name: String,
+  val sourceFile: ResourceSourceFile,
+  visibility: ResourceVisibility
+) : BasicResourceItemBase(type, name, visibility) {
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt:37]:
```suggestion
) : BasicResourceItem(type, name, visibility) {
```

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+  private var namespaceResolver: Resolver =
+    Resolver.EMPTY_RESOLVER
+
+  override fun getNamespaceResolver() = namespaceResolver
+
+  fun setNamespaceResolver(resolver: Resolver) {
+    namespaceResolver = resolver
+  }
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt:46]: nit: ordering

https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:resource-repository/main/java/com/android/resources/base/BasicValueResourceItemBase.java;l=70?q=BasicValueResourceItemBase

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+  override fun getValue(): String? = null
+
+  override fun isFileBased() = false
+
+  override fun getRepositoryConfiguration(): RepositoryConfiguration = sourceFile.configuration
+
+  override fun getSource() = originalSource
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt:54]:
```suggestion
  override fun getSource(): PathString? = originalSource
```

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItem.kt [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ * Resource item representing a value resource, e.g. a string or a color.
+ */
+class BasicValueResourceItem(
+  type: ResourceType,
+  name: String,
+  sourceFile: ResourceSourceFile,
+  visibility: ResourceVisibility,
+  private val value: String?
+) : BasicValueResourceItemBase(type, name, sourceFile, visibility) {
+
```
Comment (jrodbx) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItem.kt:36]:
```suggestion
```

**Review by jrodbx — CHANGES_REQUESTED** [review]

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicFileResourceItem.kt [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Ported from: [BasicFileResourceItem.java](https://cs.android.com/android-studio/platform/tools/base/+/18047faf69512736b8ddb1f6a6785f58d47c893f:resource-repository/main/java/com/android/resources/base/BasicFileResourceItem.java)
+ *
+ * Resource item representing a file resource, e.g. a drawable or a layout.
+ */
+class BasicFileResourceItem(
+  type: ResourceType,
+  name: String,
+  private val configuration: RepositoryConfiguration,
```
[the remark this answers is not in the document]
Comment (kevinzheng-ap) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicFileResourceItem.kt:32]: @jrodbx `repositoryConfiguration` does not exist in super class, where is the override from?

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicFileResourceItem.kt [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  visibility: ResourceVisibility,
+  private val relativePath: String
+) : BasicResourceItemBase(type, name, visibility) {
```
[the remark this answers is not in the document]
Comment (kevinzheng-ap) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicFileResourceItem.kt:35]: @jrodbx should be extended from BasicResourceItemBase as it needs base implementations

https://cs.android.com/android-studio/platform/tools/base/+/18047faf69512736b8ddb1f6a6785f58d47c893f:resource-repository/main/java/com/android/resources/base/BasicFileResourceItem.java

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicFileResourceItem.kt [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+  override fun isFileBased() = true
+
+  override fun getReference() = null
+
+  override fun getRepositoryConfiguration() = configuration
+
```
[the remark this answers is not in the document]
Comment (kevinzheng-ap) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicFileResourceItem.kt:41]: @jrodbx It is abstract function in BasicResourceItemBase

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItem.kt (hunk 1 above)
Comment (kevinzheng-ap) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItem.kt:26]: @jrodbx As discussed, I will keep the existing app.cash.paparazzi.internal.resources.BasicResourceItem for now as it has been integrated with `RepositoryLoader`.  Will delete it and migrate to this new one when overall porting is done

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItemBase.kt (hunk 2 above)
Comment (kevinzheng-ap) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicResourceItemBase.kt:33]: ditto

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt (hunk 3 above)
Comment (kevinzheng-ap) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicValueResourceItemBase.kt:37]: ditto

**Review by jrodbx — APPROVED** [review]
lgtm!

**Pull request merged**